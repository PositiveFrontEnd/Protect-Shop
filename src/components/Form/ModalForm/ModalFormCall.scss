.modal__content-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.form__modal__box {
  width: 320px;
  margin-top: 20px;
  margin-left: 40px;

  form {
    width: 100%;

    .form__field-modal {
      .svg__error-valid__modal {
        bottom: 42px;
        left: -30px;
      }

      .svg__ok-valid__modal {
        bottom: 20px;
        left: -30px;
      }

      .form__little__box {
        .form__input-modal {
          margin-top: 8px;
          margin-bottom: 8px;
          padding-top: 14px;
          padding-bottom: 12px;
          color: var(--fild-type-color);
          font-size: 14px;
          background-color: var(--white-light);
          border: 1px solid var(--black);
          border-radius: 12px;
        }
      }
    }

    .error__message-modal {
      margin-top: 3px;
    }

    .select__wrapper {
      margin-bottom: 0;
      color: var(--fild-type-color);
      font-size: 16px;
      .select__input-modal {
        font-weight: 400;
        width: 88%;
        margin-top: 8px;
        margin-bottom: 8px;
        padding-top: 14px;
        padding-bottom: 12px;
        font-size: 14px;
        background-color: var(--white-light);
        border: 1px solid var(--black);
        border-radius: 12px;
        appearance: none;
        -webkit-appearance: none;
        -moz-appearance: none;
        background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="%23000000" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" width="20" height="20"><polyline points="6 9 12 15 18 9"/></svg>');
        background-repeat: no-repeat;
        background-position: 252px center;
        select {
          padding: 0 9px;
         
          option {
            color: var(--Fild-type-color);
            font-size: 16px;
            line-height: 16px;
          }
        }
      }
    }

    .input__text {
      input {
        height: 72px;
        vertical-align: 1;
      }
    }

    .button__submit__modal-call {
      width: 88%;
      margin-top: 0;
    }
  }

  @media (width >= 768px) {
    width: 100%;
    max-width: 450px;
    margin-left: 60px;
    form {
      .select__wrapper {
        .select__input-modal {
          background-position: 360px center;
        }
      }
    }
  }

  @media (width >= 1024px) {
    max-width: 507px;
    margin-left: 80px;

    form {
      .select__wrapper {
        .select__input-modal {
          background-position: 410px center;
        }
      }
      .button__submit__modal-call {
        max-width: 447px;
      }
    }
  }
}
