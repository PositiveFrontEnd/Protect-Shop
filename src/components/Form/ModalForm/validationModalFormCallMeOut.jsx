import React from "react";
import * as Yup from "yup";

const validationModal = Yup.object({
  name: Yup.string()
    .matches(/^[a-zA-Zа-яА-ЯґҐєЄіІїЇ' ]+$/u, "Cannot have numeric values ")
    .min(2, "Last Name too short")
    .required("Required field"),
  email: Yup.string()
    .matches(/\S+@\S+\.\S+/, "Enter a valid email address")
    .min(5, "Address too short")
    .max(30, "Address too long ")
    .required("Required field"),
  phone: Yup.string()
    .matches(/^\+?[0-9]+(?:[\s\-]?[0-9]+)*$/, "Must only be numbers")
    .min(7, "phone too short")
    .max(14, "phone too long ")
    .required("Required field"),
  text: Yup.string()
    .min(5, 'Comment should be at least 5 characters long')
    .max(500, 'Comment is too long, maximum 500 characters')
});

export default validationModal;
