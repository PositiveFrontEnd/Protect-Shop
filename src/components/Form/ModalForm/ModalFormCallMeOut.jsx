import React, { useEffect } from "react";
import { Formik, Form } from "formik";
import Input from "../Inputs/Input";
import SelectInput from "../Inputs/Select";
import validationModal from "./validationModalFormCallMeOut";
import Button from "../../Button/Button";
import "./ModalFormCall.scss";
import { useDispatch, useSelector } from "react-redux";
import { actionNewLetter } from "../../../store/messageSlice";
import PropTypes from "prop-types";
import { selectorAddLetter } from "../../../store/selectors";

const ModalFormCallMeOut = () => {
  const dispatch = useDispatch();

  return (
    <Formik
      initialValues={{
        name: "",
        phone: "",
        email: "",
        question: "",
        letter: "",
      }}
      onSubmit={(values) => {
        dispatch(actionNewLetter(values));
      }}
      validationSchema={validationModal}
    >
      {({ errors, touched }) => (
        <div className="form__box form__modal__box">
          <Form>
            <Input
              className="mb-3"
              label=""
              name="name"
              placeholder="YOUR NAME"
              error={errors.name}
              touched={touched.name}
            />
            <Input
              className="mb-3"
              label=""
              name="phone"
              placeholder="YOUR PHONE"
              error={errors.phone}
              touched={touched.phone}
            />
            <Input
              className="mb-3"
              label=""
              name="email"
              placeholder="YOUR EMAIL"
              error={errors.email}
              touched={touched.email}
            />
            <SelectInput
              label=""
              name="question"
              className="mb-3"
              options={[
                { value: "my question", label: "YOUR QUESTION" },
                { value: "Product Information", label: "Product Information" },
                { value: "Shipping and Payment", label: "Shipping and Payment" },
                { value: "Returns and Exchanges", label: "Returns and Exchanges" },
                { value: "Security and Privacy", label: "Security and Privacy" },
              ]}
            />
            <Input
              className="mb-3 input__text"
              label=""
              name="letter"
              placeholder="TEXT"
              error={errors.text}
              touched={touched.text}
            />
            <Button
              children="SUBMIT"
              black
              type="submit"
              className="button__submit button__submit__modal-call"
            />
          </Form>
        </div>
      )}
    </Formik>
  );
};

ModalFormCallMeOut.propTypes = {
  isOpen: PropTypes.func,
};

export default ModalFormCallMeOut;
