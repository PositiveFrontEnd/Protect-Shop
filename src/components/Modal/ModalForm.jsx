import React from "react";
import Modal from "./ModalComponents/Modal";
import ModalWrapper from "./ModalComponents/ModalWrapper";
import ModalHeader from "./ModalComponents/ModalHeader";
import ModalClose from "./ModalComponents/ModalClose";
import ModalBody from "./ModalComponents/ModalBody";
import SvgForm from "./ModalSvg/svgForm.svg?react";
import PropTypes from 'prop-types'
import ModalFormCallMeOut from "../Form/ModalForm/ModalFormCallMeOut";

const ModalForm = ({ isOpen, onclick }) => {
    return (
        <>
            <ModalWrapper isOpen={isOpen}>
                <Modal>
                    <ModalHeader>
                        <div className="header__title">
                            <SvgForm />
                            <p className="title__text">Call me out</p>
                        </div>
                        <ModalClose onclick={onclick} />
                    </ModalHeader>
                    <ModalBody>
                        <p className="modal__body-title title__CallOut">If you still have questions, leave your contacts and our manager will call you</p>
                        <ModalFormCallMeOut isOpen={isOpen} />
                    </ModalBody>
                </Modal>
            </ModalWrapper>
        </>
    )
}

ModalForm.defaultProps = {
    onclick: () => { },
    isOpen: () => { }
}

ModalForm.propTypes = {
    onclick: PropTypes.func,
    isOpen: PropTypes.func,
}


export default ModalForm