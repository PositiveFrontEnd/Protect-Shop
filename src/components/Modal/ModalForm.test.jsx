import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import ModalForm from "./ModalForm";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  ModalForm", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {

        };
        const data = {
            price: 2134,
            onclick: () => { },
            isOpen: () => { },
        };
        const store = mockStore(initialState);
        const modalForm = render(
            <Provider store={store}>
                <BrowserRouter>
                    <ModalForm data={data} />
                </BrowserRouter>
            </Provider>
        );

        expect(modalForm).toMatchSnapshot();
    });
});