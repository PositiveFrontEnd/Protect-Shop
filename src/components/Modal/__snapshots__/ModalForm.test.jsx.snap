// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  ModalForm > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal__wrapper"
      >
        <div
          class="modal"
        >
          <div
            class="modal__header"
          >
            <div
              class="header__title"
            >
              <svg
                class="header__svg"
                fill="none"
                viewBox="0 0 16 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.5261 1.8856C15.7864 1.62525 15.7864 1.20314 15.5261 0.942787L15.0547 0.471385C14.7943 0.211035 14.3722 0.211036 14.1119 0.471385L11.2834 3.29983L10.1049 2.12127C9.9142 1.93061 9.62746 1.87357 9.37833 1.97676C9.1292 2.07995 8.9668 2.32303 8.9668 2.59268V6.36391C8.9668 6.73213 9.26526 7.0306 9.63346 7.0306H13.4047C13.6743 7.0306 13.9175 6.86813 14.0206 6.61903C14.1238 6.36992 14.0668 6.08317 13.8761 5.89251L12.6977 4.71404L15.5261 1.8856Z"
                  fill="#FEFEFE"
                />
                <path
                  d="M1.33791 3.02769C1.27065 2.07483 2.07752 1.33325 3.03079 1.33325H5.07135C5.56069 1.33325 6.00203 1.62809 6.19023 2.08072L7.04351 4.13289C7.25138 4.6328 7.10385 5.21021 6.68185 5.54853L6.14994 5.97495C5.91257 6.16524 5.83187 6.49338 5.97261 6.76307C6.69591 8.14934 7.76491 9.32581 9.06491 10.1777C9.33931 10.3575 9.70178 10.286 9.90651 10.0295L10.4608 9.33527C10.7984 8.91241 11.3746 8.76454 11.8734 8.97287L13.9214 9.82794C14.3731 10.0165 14.6673 10.4588 14.6673 10.9491V11.6577H14.6399V12.9665C14.6399 13.9217 13.8999 14.7303 12.949 14.6626C6.86871 14.2301 1.76896 9.13441 1.33791 3.02769Z"
                  fill="#FEFEFE"
                />
              </svg>
              <p
                class="title__text"
              >
                Call me out
              </p>
            </div>
            <button
              class="header__button"
              type="button"
            >
              <svg
                fill="black"
                viewBox="0 0 16 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                  fill="white"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <div
            class="modal__container"
          >
            <div
              class="modal__content modal__content-form"
            >
              <p
                class="modal__body-title title__CallOut"
              >
                If you still have questions, leave your contacts and our manager will call you
              </p>
              <div
                class="form__box form__modal__box"
              >
                <form
                  action="#"
                  name="<input type="text" placeholder="YOUR NAME" class="form__input form__input-modal" value="" name="name">"
                >
                  <label
                    class="form__field form__field-modal form__field-registr  mb-3"
                  >
                    <div
                      class="form__little__box "
                    >
                      <p
                        class="form__title"
                      />
                      <input
                        class="form__input form__input-modal"
                        name="name"
                        placeholder="YOUR NAME"
                        type="text"
                        value=""
                      />
                    </div>
                  </label>
                  <label
                    class="form__field form__field-modal form__field-registr  mb-3"
                  >
                    <div
                      class="form__little__box "
                    >
                      <p
                        class="form__title"
                      />
                      <input
                        class="form__input form__input-modal"
                        name="phone"
                        placeholder="YOUR PHONE"
                        type="text"
                        value=""
                      />
                    </div>
                  </label>
                  <label
                    class="form__field form__field-modal form__field-registr  mb-3"
                  >
                    <div
                      class="form__little__box "
                    >
                      <p
                        class="form__title"
                      />
                      <input
                        class="form__input form__input-modal"
                        name="email"
                        placeholder="YOUR EMAIL"
                        type="text"
                        value=""
                      />
                    </div>
                  </label>
                  <label
                    class="select__label select__modal  mb-3"
                  >
                    <div
                      class="select__wrapper"
                    >
                      <span
                        class="select__label__text"
                      />
                      <select
                        class="select__input select__input-modal"
                        name="question"
                      >
                        <option
                          label="YOUR QUESTION"
                          value="my question"
                        />
                        <option
                          label="Product Information"
                          value="Product Information"
                        />
                        <option
                          label="Shipping and Payment"
                          value="Shipping and Payment"
                        />
                        <option
                          label="Returns and Exchanges"
                          value="Returns and Exchanges"
                        />
                        <option
                          label="Security and Privacy"
                          value="Security and Privacy"
                        />
                      </select>
                    </div>
                  </label>
                  <label
                    class="form__field form__field-modal form__field-registr  mb-3 input__text"
                  >
                    <div
                      class="form__little__box "
                    >
                      <p
                        class="form__title"
                      />
                      <input
                        class="form__input form__input-modal"
                        name="letter"
                        placeholder="TEXT"
                        type="text"
                        value=""
                      />
                    </div>
                  </label>
                  <button
                    class="button-black button__submit button__submit__modal-call"
                    type="submit"
                  >
                    SUBMIT
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal__wrapper"
    >
      <div
        class="modal"
      >
        <div
          class="modal__header"
        >
          <div
            class="header__title"
          >
            <svg
              class="header__svg"
              fill="none"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5261 1.8856C15.7864 1.62525 15.7864 1.20314 15.5261 0.942787L15.0547 0.471385C14.7943 0.211035 14.3722 0.211036 14.1119 0.471385L11.2834 3.29983L10.1049 2.12127C9.9142 1.93061 9.62746 1.87357 9.37833 1.97676C9.1292 2.07995 8.9668 2.32303 8.9668 2.59268V6.36391C8.9668 6.73213 9.26526 7.0306 9.63346 7.0306H13.4047C13.6743 7.0306 13.9175 6.86813 14.0206 6.61903C14.1238 6.36992 14.0668 6.08317 13.8761 5.89251L12.6977 4.71404L15.5261 1.8856Z"
                fill="#FEFEFE"
              />
              <path
                d="M1.33791 3.02769C1.27065 2.07483 2.07752 1.33325 3.03079 1.33325H5.07135C5.56069 1.33325 6.00203 1.62809 6.19023 2.08072L7.04351 4.13289C7.25138 4.6328 7.10385 5.21021 6.68185 5.54853L6.14994 5.97495C5.91257 6.16524 5.83187 6.49338 5.97261 6.76307C6.69591 8.14934 7.76491 9.32581 9.06491 10.1777C9.33931 10.3575 9.70178 10.286 9.90651 10.0295L10.4608 9.33527C10.7984 8.91241 11.3746 8.76454 11.8734 8.97287L13.9214 9.82794C14.3731 10.0165 14.6673 10.4588 14.6673 10.9491V11.6577H14.6399V12.9665C14.6399 13.9217 13.8999 14.7303 12.949 14.6626C6.86871 14.2301 1.76896 9.13441 1.33791 3.02769Z"
                fill="#FEFEFE"
              />
            </svg>
            <p
              class="title__text"
            >
              Call me out
            </p>
          </div>
          <button
            class="header__button"
            type="button"
          >
            <svg
              fill="black"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                fill="white"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal__container"
        >
          <div
            class="modal__content modal__content-form"
          >
            <p
              class="modal__body-title title__CallOut"
            >
              If you still have questions, leave your contacts and our manager will call you
            </p>
            <div
              class="form__box form__modal__box"
            >
              <form
                action="#"
                name="<input type="text" placeholder="YOUR NAME" class="form__input form__input-modal" value="" name="name">"
              >
                <label
                  class="form__field form__field-modal form__field-registr  mb-3"
                >
                  <div
                    class="form__little__box "
                  >
                    <p
                      class="form__title"
                    />
                    <input
                      class="form__input form__input-modal"
                      name="name"
                      placeholder="YOUR NAME"
                      type="text"
                      value=""
                    />
                  </div>
                </label>
                <label
                  class="form__field form__field-modal form__field-registr  mb-3"
                >
                  <div
                    class="form__little__box "
                  >
                    <p
                      class="form__title"
                    />
                    <input
                      class="form__input form__input-modal"
                      name="phone"
                      placeholder="YOUR PHONE"
                      type="text"
                      value=""
                    />
                  </div>
                </label>
                <label
                  class="form__field form__field-modal form__field-registr  mb-3"
                >
                  <div
                    class="form__little__box "
                  >
                    <p
                      class="form__title"
                    />
                    <input
                      class="form__input form__input-modal"
                      name="email"
                      placeholder="YOUR EMAIL"
                      type="text"
                      value=""
                    />
                  </div>
                </label>
                <label
                  class="select__label select__modal  mb-3"
                >
                  <div
                    class="select__wrapper"
                  >
                    <span
                      class="select__label__text"
                    />
                    <select
                      class="select__input select__input-modal"
                      name="question"
                    >
                      <option
                        label="YOUR QUESTION"
                        value="my question"
                      />
                      <option
                        label="Product Information"
                        value="Product Information"
                      />
                      <option
                        label="Shipping and Payment"
                        value="Shipping and Payment"
                      />
                      <option
                        label="Returns and Exchanges"
                        value="Returns and Exchanges"
                      />
                      <option
                        label="Security and Privacy"
                        value="Security and Privacy"
                      />
                    </select>
                  </div>
                </label>
                <label
                  class="form__field form__field-modal form__field-registr  mb-3 input__text"
                >
                  <div
                    class="form__little__box "
                  >
                    <p
                      class="form__title"
                    />
                    <input
                      class="form__input form__input-modal"
                      name="letter"
                      placeholder="TEXT"
                      type="text"
                      value=""
                    />
                  </div>
                </label>
                <button
                  class="button-black button__submit button__submit__modal-call"
                  type="submit"
                >
                  SUBMIT
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
